internal/server/handlers/config.go

```diff
@@ -51,59 +61,87 @@ func (s *Service) updateConfig(w http.ResponseWriter, r *http.Request) {
 
 	newConfig, err := dto.NewConfigFromReader(r.Body, dto.JSON)
 	if err != nil {
-		// We won't log config as it is not secure.
 		hLogger.Error("failed to decode new configuration",
 			slog.Any("error", err),
 		)
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	s.config, err = newConfig.ToModel()
+
+	newConfigModel, err := newConfig.ToModel()
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	err = s.changeConfig(r.Context(), func(config *model.Config) error {
+		config.CopyFrom(newConfigModel)
+		return nil
+	})
 
-	err = s.configurationManager.WriteConfiguration(r.Context(), s.config)
 	if err != nil {
-		// We won't log config as it is not secure.
-		hLogger.Error("failed to update configuration",
+		hLogger.Error("failed to apply config",
 			slog.Any("error", err),
 		)
-		http.Error(w, err.Error(), http.StatusBadRequest)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
 	w.WriteHeader(http.StatusOK)
 }
 
-// ApplyConfig
+// ApplyConfig  read and apply configuration from file.
 // @Summary     Applies the configuration for the service.
 // @ID          applyConfig
 // @Tags        Configuration
 // @Router      /v1/config/apply [post]
 // @Accept      json
 // @Success     200
 // @Failure     400 {string} string
-func (s *Service) ApplyConfig(w http.ResponseWriter, _ *http.Request) {
+func (s *Service) ApplyConfig(w http.ResponseWriter, r *http.Request) {
 	hLogger := s.logger.With(slog.String("handler", "ApplyConfig"))
 
-	handlers, err := service.ApplyNewConfig(s.scheduler, s.config, s.backupBackends, s.clientManger)
+	config, err := s.configurationManager.Read(r.Context())
+	if err != nil {
+		hLogger.Error("failed to read config",
+			slog.Any("error", err),
+		)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = s.changeConfig(r.Context(), func(c *model.Config) error {
```

Comment:
`changeConfig` calls to `configurationManager.Write` which is unnecessary and also not supported for configuration managers other than local.